src/main/kotlin/com/exactpro/th2/common/utils/message/transport/MessageBatcher.kt

```diff
@@ -68,9 +71,12 @@ class MessageBatcher(
         private var batch = newBatch()
         private var future: Future<*> = CompletableFuture.completedFuture(null)
 
-        fun add(message: Message.Builder<*>) = lock.withLock {
-            message.idBuilder().setTimestamp(Instant.now())
-            batch.addGroup(message.build().toGroup())
+        fun add(builder: Message.Builder<*>) = lock.withLock {
+            builder.idBuilder().setTimestamp(Instant.now())
+            val message: Message<*> = builder.build()
+
+            beforeAddMessage(message)
+            batch.addGroup(message.toGroup())
```

Comment:
note: I am not sure, that this is a good approach. The purpose of `beforeAddMessage` is not clear.

In my opinion, it would be better if we keep all the `send` calls here and in `close` method. How about we add a method `tryAddToBatch` that will return `true` if message was added to batch and `false` if the batch is full. In the latter case, we would explicitly call the `send` method here

Reply:
Corrected the code, but I haven't found appropriate names for method )